x/payment/keeper/keeper.go

```diff
@@ -35,15 +35,12 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 //  MsgPayForData moves a user's coins to the module address and burns them.
 func (k Keeper) PayForData(goCtx context.Context, msg *types.MsgPayForData) (*types.MsgPayForDataResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	ctx.GasMeter().ConsumeGas(msg.MessageSize, payForDataGasDescriptor)
```

Comment:
Qs: 
- Can a block proposer still include larger blobs than the size that was committed to in the PFD? Is there a test somewhere for that?
- Can a block proposer include a blob without a PFD? (my understanding is no; just double checking)

Reply:
Isn't including larger blobs and including blob without PFD isomorphic?

Reply:
After we add the message inclusion check, it will not be possible without 2/3s of the voting power, as validators recompute the commitment for each message and compare them with those in the pfds

Reply:
> Isn't including larger blobs and including blob without PFD isomorphic?

Yeah, good point. 

> After we add the message inclusion check, it will not be possible without 2/3s of the voting power, as validators recompute the commitment for each message and compare them with those in the pfds

That is in line with the honest majority assumption for state. So I think that is good enough.